### PR TITLE
release: v0.2.0

### PR DESCRIPTION
**Pull request by ymjiang**
This PR is for the official release of `v0.2.0`

### CODE REVIEW

**Comment by byronyi** [comment]
Congrats! It will be better to add a `CHANGELOG.rst` file to mention the important changes from last version.

**Comment by ymjiang** [comment]
Added a changelog to indicate the major changes. @byronyi

**Comment by ymjiang** [comment]
@bobzhuyb Please take a review. Major changes:
- Uploaded pip package: `pip3 install byteps`
- Updated all docker images.
- Fixed stale documents.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Changelog for BytePS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+0.2.0 (2020-02)
+------------------
+* Largely improve RDMA performance by enforcing page aligned memory and adding IPC support. Now support colocating servers and workers without sacrificing much performance.
```
Comment (bobzhuyb) [inline CHANGELOG.rst]: page aligned and IPC support can be two separate items.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+* Fix a hanging bug in BytePS server.
+* Fix RDMA fork problem caused by multi-processing (e.g., data loaders).
```
Comment (bobzhuyb) [inline CHANGELOG.rst]: Fix RDMA-related segmentation fault problem during fork() (e.g., used by PyTorch data loader).

File: CHANGELOG.rst [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+* New feature: Enable mixing use of colocate and non-colocate servers, along with a smart tensor allocation strategy.
+* Add support for pip install.
```
Comment (bobzhuyb) [inline CHANGELOG.rst]: Add the pip install command here.

File: README.md [hunk 4]
```diff
@@ -41,21 +46,30 @@ BytePS also incorporates many acceleration techniques such as hierarchical strat
 
 We provide a [step-by-step tutorial](docs/step-by-step-tutorial.md) for you to run benchmark training tasks. The simplest way to start is to use our [docker images](docker). Refer to [Documentations](docs) for how to [launch distributed jobs](docs/running.md) and more [detailed configurations](docs/env.md). After you can start BytePS, read [best practice](docs/best-practice.md) to get the best performance.
 
-Below, we explain how to build and run BytePS by yourself. BytePS assumes that you have already installed one or more of the following frameworks: TensorFlow / PyTorch / MXNet. BytePS depends on CUDA and NCCL, and requires gcc>=4.9. If you are working on CentOS/Redhat and have gcc<4.9, you can try `yum install devtoolset-7` before everything else.
+Below, we explain how to install BytePS by yourself. There are two options.
+
+### Build from pip
```
Comment (bobzhuyb) [inline README.md]: Build from pip --> Install by pip

File: README.md [hunk 5]
```diff
@@ -41,21 +46,30 @@ BytePS also incorporates many acceleration techniques such as hierarchical strat
...
 ```
 
-Notes:
-- For best compatibility, please pin your gcc to 4.9 before building, [here](https://github.com/bytedance/byteps/blob/master/docker/Dockerfile.pytorch#L72-L80) is an example.
-- You may set `BYTEPS_USE_RDMA=1` to install with RDMA support. Before this, make sure your RDMA drivers have been properly installed and tested.
+Notes for above two options:
+- BytePS assumes that you have already installed one or more of the following frameworks: TensorFlow / PyTorch / MXNet. 
+- BytePS depends on CUDA and NCCL. You should specify the NCCL path with `export BYTEPS_NCCL_HOME=/path/to/nccl`. By default it points to `/usr/local/nccl`.
+- The installation requires gcc>=4.9. If you are working on CentOS/Redhat and have gcc<4.9, you can try `yum install devtoolset-7` before everything else. In general, we recommend using gcc 4.9 for best compatibility ([example](https://github.com/bytedance/byteps/blob/master/docker/Dockerfile.pytorch#L72-L80)).
+- RDMA support: During setup, the script will automatically detect the RDMA header file. If you want to use RDMA, make sure your RDMA environment has been properly installed and tested before install.
```
Comment (bobzhuyb) [inline README.md]: Can you be more specific, e.g., on Ubuntu, users need to have xxx..

File: docs/best-practice.md [hunk 6]
```diff
@@ -13,15 +13,23 @@ If you have NVLinks, leave `BYTEPS_PCIE_SWITCH_SIZE` unmodified. If you don't kn
...
+To address this, you can try running more BytePS server instances on the server machines. For example, you can try running two server instances per server machines. This effectively doubles the number of TCP connections and should be sufficient for 25Gbps networks. For 40Gbps/50Gbps networks, you need three server instances per server machine, and so on. 
+
+### No additional CPU servers
+
+When you don't have additional CPU servers, then for each physical machine, you should launch a worker and a server process. We call this *co-locate* mode, and the resource consumption is the same with Horovod (no additional servers). 
+
+If you are using TCP, you will probably get near-identical performance with Horovod-TCP. However, if you are using RDMA, you can set `BYTEPS_ENABLE_IPC=1` to enable the IPC communication between the co-located worker and server. And eventually you will get higher end-to-end performance than Horovod.
```
Comment (bobzhuyb) [inline docs/best-practice.md]: Why don't we enable IPC by default, for RDMA?

File: CHANGELOG.rst [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+
```
Comment (bobzhuyb) [inline CHANGELOG.rst]: bpslaunch is a new feature, too

**Pull request merged**